test/test_thruster_feedback.rb

```diff
@@ -0,0 +1,214 @@
+  end
+
+  it 'automatically setting control_modes' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml")
+    
+    aux = thrusters_feedback.control_modes
+    aux.clear
+    thrusters_feedback.control_modes = aux 
+      
+    thrusters_feedback.configure
+  end
+
+  it 'pos and neg coefficients have different sizes' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml",['default', 'thruster_coeff_pos_size_3'], true)
+    
+    aux = thrusters_feedback.control_modes
+    aux.clear
+    thrusters_feedback.control_modes = aux 
+      
+    assert_raises(Orocos::StateTransitionFailed) { thrusters_feedback.configure }
+  end
+
+  it 'control modes set to EFFORT' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml",['default', 'control_modes_effort'], true)
+    
+    assert_raises(Orocos::StateTransitionFailed) { thrusters_feedback.configure }
+  end
+
+  it 'control mode equals to RAW but no thruster voltage was set' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml")
+    
+    thrusters_feedback.thruster_voltage = 0
+    
+    assert_raises(Orocos::StateTransitionFailed) { thrusters_feedback.configure }
+  end
+
+  it 'wrong number of thrusters input' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml")
+    
+    thrusters_feedback.configure  
+    thrusters_feedback.start  
+      
+    sample = thrusters_feedback.cmd_in.new_sample
+    
+    # 3 thrusters inputs were sent and 2 are expected
+    thruster = Types::Base::JointState.new
+    thruster.effort = 3
+    sample.elements = [thruster, thruster, thruster]
+    cmd_in.write sample
+
+    assert_state_change(thrusters_feedback) { |s| s == :UNEXPECTED_THRUSTER_INPUT } 
+        
+  end
+  
+  it 'thruster input not set' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml")
+    
+    thrusters_feedback.configure  
+    thrusters_feedback.start  
+
+    sample = thrusters_feedback.cmd_in.new_sample
+      
+    # the effort field was not set
+    thruster = Types::Base::JointState.new
+    sample.elements = [thruster, thruster]
+    cmd_in.write sample
+
+    assert_state_change(thrusters_feedback) { |s| s == :UNSET_THRUSTER_INPUT } 
+        
+  end
+  
+  it 'testing positive speed calculated value' do
+
+    thrusters_feedback.apply_conf_file("auv_control::ThrustersFeedback.yml",['default', 'control_modes_speed'], true)
+    
+    thrusters_feedback.configure  
+    thrusters_feedback.start  
+
+    sample = thrusters_feedback.cmd_in.new_sample
+      
+    thruster1 = Types::Base::JointState.new
+    thruster2 = Types::Base::JointState.new
+    thruster1.speed = 457
+    thruster2.speed = 789
+    sample.elements = [thruster1, thruster2]
+    cmd_in.write sample
+
+    data = assert_has_one_new_sample cmd_out, 1
+    
+    assert (data.elements[0].effort - 13.99).abs < 0.001
```

Comment:
You can keep it that way, but there's [assert_in_delta](http://ruby-doc.org/stdlib-1.9.2/libdoc/minitest/unit/rdoc/MiniTest/Assertions.html#method-i-assert_in_delta) and [assert_in_epsilon](http://ruby-doc.org/stdlib-1.9.2/libdoc/minitest/unit/rdoc/MiniTest/Assertions.html#method-i-assert_in_epsilon)

Their plus is that they give a much nicer error message if the assertion fails.

Note that most test frameworks have this kind of test assertions to compare floats.